TriblerGUI/widgets/trustpage/js/data_processor.js

```diff
@@ -0,0 +1,326 @@
+ * Get the focus node object and its public key.
+ *
+ * @param {GraphResponseData} response - The response
+ * @param {Object} interim - The interim result (expects interim.nodes)
+ * @returns {{focus_node: GraphNode, focus_pk: String}}
+ */
+function focusNodePublicKey(response, interim) {
+
+    return {
+        focus_node: find(interim.nodes, "public_key", response.focus_node),
+        focus_pk: response.focus_node
+    };
+}
+
+/**
+ * Add the neighbors to each node. (Manipulates the interim)
+ *
+ * @param {GraphResponseData} response - The response
+ * @param {Object} interim - The interim result (expects interim.links, interim.nodes)
+ * @returns {{}} - An empty object, since this manipulates the interim.
+ */
+function addNeighborsToNodes(response, interim) {
+
+    interim.nodes.forEach(function(node){
+        console.log(11,node.public_key)
+        node.neighbors = [];
+    });
+
+    interim.links.forEach(function (link) {
+        console.log(22,link.source.public_key)
+        link.source.neighbors.push(link.target);
+        link.target.neighbors.push(link.source);
+    });
+
+    console.log("NB", interim.nodes[0].neighbors);
+
+    return {};
+}
+
+/**
+ * Sort the nodes array in-place in the result object ascending on up+down.
+ * (Manipulates the interim)
+ *
+ * @param {GraphResponseData} response - The response
+ * @param {Object} interim - The interim result (expects interim.nodes)
+ * @returns {{}} - An empty object, since this manipulates the interim.
+ */
+function sortNodes(response, interim) {
+
+    interim.nodes.sort(function (nodeOne, nodeTwo) {
+        return nodeOne.total_up + nodeOne.total_down - nodeTwo.total_up - nodeTwo.total_down;
+    });
+
+    return {};
+}
+
+/**
+ * Add the page min and max page rank of the set of nodes.
+ *
+ * @param {GraphResponseData} response - The response
+ * @returns {{min_page_rank: number, max_page_rank: number}}
+ */
+function addPageRank(response) {
+
+    var sortedPageRank = response.nodes.map(function (node) {
+        return node.page_rank;
+    }).sort();
+
+    return {
+        min_page_rank: sortedPageRank[0],
+        max_page_rank: sortedPageRank[sortedPageRank.length - 1],
+    };
+}
+
+/**
+ * Returns the first object in a list of objects that matches a given key-value pair.
+ *
+ * @param {Object[]} list - The list of objects
+ * @param {String} key    - The key to look at
+ * @param {*} val         - The value to match for
+ * @returns {Object|null} - The matching object or null if no match
+ */
+function find(list, key, val) {
+    return list.find(function (item) {
+        return item[key] === val;
+    });
+}
+
+/**
+ * Returns a concatenation of arrays returned from mapping an array over a callback.
+ *
+ * @param {Array} arr         - The array to reduce
+ * @param {Function} callback - The callback to apply, must return an array
+ * @returns {Array} arr       - The concatenated result of all calls to callback
+ */
+function flatMap(arr, callback) {
```

Comment:
Why do you write a new function for what is essentially just the Array.map function?